### PR TITLE
Add day (non night vision) version of the vector 21

### PR DESCRIPTION
**Pull request by PabstMirror**
**When merged this pull request will:**
 - Add day (non night vision) version of the vector 21
 - Add "Nite" to the name of the current NV capable vector

Ref http://www.vectronix.us/handhelds/vector-family-all-1

### CODE REVIEW

**Comment by esteldunedain** [comment]
:+1:

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 1]
```diff
@@ -88,7 +88,7 @@ if (GVAR(ammoTemperatureEnabled) || GVAR(barrelLengthInfluenceEnabled)) then {
 
 _bulletTraceVisible = false;
 if (GVAR(bulletTraceEnabled) && cameraView == "GUNNER") then {
-    if (currentWeapon ACE_player in ["ACE_Vector", "Binocular", "Rangefinder", "Laserdesignator"]) then {
+    if (currentWeapon ACE_player in ["ACE_Vector", "ACE_VectorDay", "Binocular", "Rangefinder", "Laserdesignator"]) then {
```
Comment (Glowbal) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: this is ok for now, but we should really avoid hardcoding this list of binoculars

File: addons/dagr/functions/fnc_menuInit.sqf [hunk 2]
```diff
@@ -522,7 +522,7 @@ GVAR(menuRun) = true;
                     (__dsp displayCtrl __mainText) ctrlSetText "Connecting...";
                 };
                 if (ACE_time - GVAR(busyTimer) > 0.6) then {
-                    if (ACE_player hasWeapon "ACE_Vector") then {
+                    if ((ACE_player hasWeapon "ACE_Vector") || {ACE_player hasWeapon "ACE_VectorDay"}) then {
```
Comment (Glowbal) [inline addons/dagr/functions/fnc_menuInit.sqf]: likewise here, we should avoid hardcoding our vector and use a more flexible approach

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf (hunk 1 above)
Comment (jonpas) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: `currentWeapon ACE_player == binocular ACE_player`

**Comment by PabstMirror** [comment]
fixed, I didn't like that static and case sensitive list either
AB bullet trace should now also be compatible with any other mod's binoculars

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 3]
```diff
@@ -88,7 +88,7 @@ if (GVAR(ammoTemperatureEnabled) || GVAR(barrelLengthInfluenceEnabled)) then {
 
 _bulletTraceVisible = false;
 if (GVAR(bulletTraceEnabled) && cameraView == "GUNNER") then {
-    if (currentWeapon ACE_player in ["ACE_Vector", "Binocular", "Rangefinder", "Laserdesignator"]) then {
+    if ((currentWeapon ACE_player == binocular ACE_player) && {binocular ACE_player != ""}) then {
```
Comment (jonpas) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: is `binocular ACE_player != ""` check necessary? If there is no binocular then first part will be false anyways.

Comment (PabstMirror) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: I was thinking of a civilian with no weapon: `"" == ""`, but I guess they wouldn't be able to be in "GUNNER" view

**Comment by Glowbal** [comment]
Very nice solution @PabstMirror :+1:

**Comment by commy2** [comment]
merge?

**Pull request merged**